doc/level0.rst

```diff
@@ -137,32 +137,37 @@ specific format, which contains a payload protected by the setup code.
   transferable secret key.
 - The encryption algorithm used MUST be at least as strong as 128-bit
   symmetric encryption.  It SHOULD be AES-128.  The passphrase MUST be
-  the Setup Code (see below), used under OpenPGP's salted+iterated S2K
-  algorithm.
+  the Setup Code (see below) represented as a UTF-8 string, used under
+  OpenPGP's salted+iterated S2K algorithm.
 
 The payload MIME part MAY optionally use the OpenPGP Transfer Format
-(see :ref:`openpgp-transfer-format`) for improved usability.
+(see :doc:`transfer-format`) for improved usability.
 
 Setup Code
 ++++++++++
 
 The setup code MUST be generated by the implementation itself using a
 CSPRNG, and presented directly to the user for safekeeping. It MUST
-NOT be included in the Autocrypt Setup Message, or otherwise
-transmitted over e-mail. Its format SHOULD be 24 uppercase
-alphanumeric characters, divided into six blocks of four, separated by
-dashes. The dashes are part of the secret code. This format holds 124
-bits of data, it is designed to be unambiguous, pronounceable, and
-split into blocks that can be easily kept in short term memory. For
-instance:
-
-``AB1D-E2GH-IJK3-4NOP-Q5ST-XYZ6``
-
-When this form of Setup Code is used, the ASCII-armored block in the
-payload SHOULD include an OpenPGP Armor header called
-"Autocrypt-Setup-Message" with a value of "v1", to indicate to
-receiving implementations the format of the passphrase.  This is a
-useful indicator at message import time.
+NOT be included in the cleartext of the Autocrypt Setup Message, or
+otherwise transmitted over e-mail.
+
+An Autocrypt level 0 client MUST generate a setup code as UTF-8 string
+of 24 uppercase alphanumeric characters, divided into six blocks of
+four, separated by dashes. The dashes are part of the secret
+code. This format holds about 124 bits of data. It is designed to be
+unambiguous, pronounceable, and split into blocks that can be easily
+kept in short term memory. For instance::
+
+    AB1D-E2GH-IJK3-4NOP-Q5ST-XYZ6
+
+An Autocrypt Setup Message payload that uses this structure for its
+setup code SHOULD include the following header in the outer OpenPGP
```

Comment:
Requiring an Autocrypt-Setup-Message header with some value would allow using this header to identify the Setup Message. The SHOULD in here refers to the value of the header.